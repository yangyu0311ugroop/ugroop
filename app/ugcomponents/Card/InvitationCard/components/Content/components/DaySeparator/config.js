import { INVITATION_STORE } from 'appConstants';

export const CONFIG = {
  value: {
    createdAt: ({ token, sortBy = 'updatedAt' }) => [
      INVITATION_STORE,
      'shares',
      token,
      sortBy,
    ],
    previousCreatedAt: ({ previousToken, sortBy = 'updatedAt' }) => [
      INVITATION_STORE,
      'shares',
      previousToken,
      sortBy,
    ],
  },
  setValue: {},
};
