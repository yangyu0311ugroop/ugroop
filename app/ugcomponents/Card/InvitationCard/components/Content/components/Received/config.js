import {
  INVITATION_STORE,
  TEMPLATE_MANAGEMENT_DATASTORE,
  USER_DATA_STORE,
} from 'appConstants';
import { get } from 'lodash';

export const CONFIG = {
  value: {
    shareToUserId: ({ token }) => [
      INVITATION_STORE,
      'shares',
      token,
      'shareToUserId',
    ],
    shareFromUserId: ({ token }) => [
      INVITATION_STORE,
      'shares',
      token,
      'shareFrom',
    ],
    nodeOrgId: ({ token }) => [INVITATION_STORE, 'shares', token, 'orgId'],
    nodeOrgName: ({ token }) => [INVITATION_STORE, 'shares', token, 'orgName'],
    nodeId: ({ token }) => [INVITATION_STORE, 'shares', token, 'nodeId'],
    status: ({ token }) => [INVITATION_STORE, 'shares', token, 'status'],
    role: ({ token }) => [INVITATION_STORE, 'shares', token, 'role'],
    content: ({ token }) => [INVITATION_STORE, 'shares', token, 'content'], // TODO not yet implemented
    personalMessage: ({ token }) => [
      INVITATION_STORE,
      'notifications',
      token,
      'content',
      'content',
    ],
    organisationName: ({ token }) => [
      INVITATION_STORE,
      'notifications',
      token,
      'content',
      'organisationName',
    ],

    orgId: {
      keyPath: [
        ({ token }) => [INVITATION_STORE, 'shares', token, 'shareFrom'],
        [USER_DATA_STORE, 'users'],
      ],
      getter: (shareFromUserId, users = {}) =>
        get(users, `${shareFromUserId}.orgId`),
    },
    target: {
      keyPath: [TEMPLATE_MANAGEMENT_DATASTORE, 'id'],
      getter: id => (id ? '_blank' : ''),
    },
  },
  setValue: {
    exclusion: [INVITATION_STORE, 'exclusion'],
  },
};
