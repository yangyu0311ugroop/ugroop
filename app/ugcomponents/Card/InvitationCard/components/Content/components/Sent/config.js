import { INVITATION_STORE, USER_DATA_STORE } from 'appConstants';
import { get } from 'lodash';

export const CONFIG = {
  value: {
    shareTo: ({ token }) => [INVITATION_STORE, 'shares', token, 'shareTo'],
    shareToUserId: ({ token }) => [
      INVITATION_STORE,
      'shares',
      token,
      'shareToUserId',
    ],
    nodeId: ({ token }) => [INVITATION_STORE, 'shares', token, 'nodeId'],
    status: ({ token }) => [INVITATION_STORE, 'shares', token, 'status'],
    role: ({ token }) => [INVITATION_STORE, 'shares', token, 'role'],
    content: ({ token }) => [INVITATION_STORE, 'shares', token, 'content'], // TODO not yet implemented
    personalMessage: ({ token }) => [
      INVITATION_STORE,
      'notifications',
      token,
      'content',
      'content',
    ],
    organisationName: ({ token }) => [
      INVITATION_STORE,
      'notifications',
      token,
      'content',
      'organisationName',
    ],

    orgId: {
      keyPath: [
        ({ token }) => [INVITATION_STORE, 'shares', token, 'shareToUserId'],
        [USER_DATA_STORE, 'users'],
      ],
      getter: (shareToUserId, users = {}) =>
        get(users, `${shareToUserId}.orgId`),
    },
  },
  setValue: {},
};
