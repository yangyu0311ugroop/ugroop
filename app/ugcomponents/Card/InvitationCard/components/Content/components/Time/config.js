import { INVITATION_STORE } from 'appConstants';

export const CONFIG = {
  value: {
    createdAt: ({ token, sortBy = 'updatedAt' }) => [
      INVITATION_STORE,
      'shares',
      token,
      sortBy,
    ],
  },
  setValue: {},
};
