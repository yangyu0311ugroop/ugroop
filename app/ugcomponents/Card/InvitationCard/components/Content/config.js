import { INVITATION_STORE, TEMPLATE_MANAGEMENT_VIEWSTORE } from 'appConstants';
import { INVITATION_VIEW_STORE_SELECTORS } from 'datastore/invitationStore/selectors';

export const CONFIG = {
  value: {
    show: INVITATION_VIEW_STORE_SELECTORS.show,
    showCompleted: INVITATION_VIEW_STORE_SELECTORS.showCompleted,

    sent: [INVITATION_STORE, 'fromMe'],
    received: [INVITATION_STORE, 'toMe'],
    completedToMe: [INVITATION_STORE, 'completedToMe'],
    completedFromMe: [INVITATION_STORE, 'completedFromMe'],

    seeDetail: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'seeDetail'],
  },
  setValue: {
    seeDetail: [TEMPLATE_MANAGEMENT_VIEWSTORE, 'seeDetail'],
  },
};
