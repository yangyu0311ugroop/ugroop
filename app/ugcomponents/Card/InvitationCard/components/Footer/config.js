import { INVITATION_VIEW_STORE_SELECTORS } from 'datastore/invitationStore/selectors';

export const CONFIG = {
  value: {
    showCompleted: INVITATION_VIEW_STORE_SELECTORS.showCompleted,
  },
  setValue: {
    showCompleted: INVITATION_VIEW_STORE_SELECTORS.showCompleted,
  },
};
