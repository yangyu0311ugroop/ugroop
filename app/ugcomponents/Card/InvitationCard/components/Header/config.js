import { arrays } from 'utils';
import { INVITATION_STORE, USER_PREFERENCE } from 'appConstants';
import { INVITATION_VIEW_STORE_SELECTORS } from 'datastore/invitationStore/selectors';
import { COGNITO_STORE_SELECTORS } from 'datastore/stormPathStore/selectors';
import { USERS_PREFERENCE_SELECTORS } from 'datastore/userStore/selectors';

export const USER_ID_CONFIG = {
  value: {
    userId: COGNITO_STORE_SELECTORS.myId,
  },
};
export const CONFIG = {
  value: {
    show: INVITATION_VIEW_STORE_SELECTORS.show,
    showCompleted: INVITATION_VIEW_STORE_SELECTORS.showCompleted,

    sentCount: {
      keyPath: [INVITATION_STORE, 'fromMe'],
      getter: arrays.length,
    },
    receivedCount: {
      keyPath: [INVITATION_STORE, 'toMe'],
      getter: arrays.length,
    },
    completedFromMeCount: {
      keyPath: [INVITATION_STORE, 'completedFromMe'],
      getter: arrays.length,
    },
    completedToMeCount: {
      keyPath: [INVITATION_STORE, 'completedToMe'],
      getter: arrays.length,
    },
    expanded: {
      keyPath: ({ userId }) => [
        USERS_PREFERENCE_SELECTORS.getUserPreference({
          id: userId,
          key: USER_PREFERENCE.DASH_BOARD_INVITATION,
        }),
      ],
      getter: (value = USER_PREFERENCE.BOOL_VALUE) =>
        value === USER_PREFERENCE.BOOL_VALUE,
    },
  },

  setValue: {
    show: INVITATION_VIEW_STORE_SELECTORS.show,
  },
};
