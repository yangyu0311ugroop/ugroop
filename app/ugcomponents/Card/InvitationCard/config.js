import { COGNITO_ACCOUNTSTORE, USER_PREFERENCE } from 'appConstants';
import { INVITATION_VIEW_STORE_SELECTORS } from 'datastore/invitationStore/selectors';
import { COGNITO_STORE_SELECTORS } from 'datastore/stormPathStore/selectors';
import { USERS_PREFERENCE_SELECTORS } from 'datastore/userStore/selectors';
import { USER_API, UPDATE_USER_PREFERENCE } from 'apis/constants';

export const USER_ID_CONFIG = {
  value: {
    userId: COGNITO_STORE_SELECTORS.myId,
  },
};
export const CONFIG = {
  value: {
    userId: [COGNITO_ACCOUNTSTORE, 'account', 'id'],
    showCompleted: INVITATION_VIEW_STORE_SELECTORS.showCompleted,
    open: {
      keyPath: ({ userId }) => [
        USERS_PREFERENCE_SELECTORS.getUserPreference({
          id: userId,
          key: USER_PREFERENCE.DASH_BOARD_INVITATION,
        }),
      ],
      getter: (value = USER_PREFERENCE.BOOL_VALUE) =>
        value === USER_PREFERENCE.BOOL_VALUE,
    },
  },

  setValue: {},
  isLoading: {
    updateUserPreference: [USER_API, UPDATE_USER_PREFERENCE],
  },
};
