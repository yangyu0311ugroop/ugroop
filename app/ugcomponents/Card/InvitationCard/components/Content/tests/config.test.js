import { CONFIG } from '../config';

describe('Content/config.js', () => {
  afterEach(() => jest.clearAllMocks());

  describe('Smoke Test', () => {
    it('should match snapshot', () => {
      expect(CONFIG).toMatchSnapshot();
    });
  });
});
