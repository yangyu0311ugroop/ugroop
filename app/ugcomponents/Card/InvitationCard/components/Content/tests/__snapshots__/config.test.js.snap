// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content/config.js Smoke Test should match snapshot 1`] = `
Object {
  "setValue": Object {
    "seeDetail": Array [
      "templateManagementView",
      "seeDetail",
    ],
  },
  "value": Object {
    "completedFromMe": Array [
      "invitationStore",
      "completedFromMe",
    ],
    "completedToMe": Array [
      "invitationStore",
      "completedToMe",
    ],
    "received": Array [
      "invitationStore",
      "toMe",
    ],
    "seeDetail": Array [
      "templateManagementView",
      "seeDetail",
    ],
    "sent": Array [
      "invitationStore",
      "fromMe",
    ],
    "show": [Function],
    "showCompleted": [Function],
  },
}
`;
